lib/transport/objects.ts

```diff
@@ -173,6 +183,10 @@ export class GroupWriter {
 	async write(c: GroupChunk) {
 		await this.stream.u53(c.object)
 		await this.stream.u53(c.payload.byteLength)
+		// Write object status if payload empty
+		if (c.payload.byteLength == 0) {
```

Comment:
```typescript
if (c.payload instanceof Uint8Array) {
  this.stream.write(c.payload)
} else {
  this.stream.u53(c.payload as number)
}
```

And yes, consumers of this library will have to perform this check too, but that's why making the payload nullable via a status code is a dumb idea.